network/src/node.rs

```diff
@@ -199,11 +199,11 @@ impl<S: Storage + Send + core::marker::Sync + 'static> Node<S> {
             let node_clone = self.clone();
 
             let known_network_task = task::spawn(async move {
-                loop {
-                    // Should always be present since we check for it before this block.
-                    if let Some(known_network) = node_clone.known_network() {
-                        known_network.update().await
-                    }
+                let known_network = node_clone.known_network().unwrap();
```

Comment:
```suggestion
                // Safe since we check the presence of `known_network`. 
                let known_network = node_clone.known_network().unwrap();
```

Reply:
sure, why not; also, I'd just have gone with `if let Some` earlier on, but we're spawning a task here, and `known_network()` returns a reference, so it wasn't viable